pkg/extract/freebsd/freebsd.go

```diff
@@ -55,10 +66,144 @@ func Extract(args string, opts ...Option) error {
+
+	rs := make([]reference.Reference, 0, reflen)
+	for _, u := range fetched.References.URL {
+		rs = append(rs, reference.Reference{
+			Source: "vuxml.freebsd.org",
+			URL:    u,
+		})
+	}
+	for _, a := range fetched.References.FreebsdSA {
+		rs = append(rs, reference.Reference{
+			Source: "vuxml.freebsd.org",
+			URL:    fmt.Sprintf("https://www.freebsd.org/security/advisories/FreeBSD-%s.asc", a),
+		},
+		)
+	}
+	for _, a := range fetched.References.FreebsdPR {
+		rs = append(rs, reference.Reference{
+			Source: "vuxml.freebsd.org",
+			URL:    fmt.Sprintf("https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=%s", strings.TrimPrefix(a, "ports/")),
+		})
+	}
+	for _, m := range fetched.References.Mlist {
+		rs = append(rs, reference.Reference{
+			Source: "vuxml.freebsd.org",
+			URL:    m.Text,
+		})
+	}
+	for _, b := range fetched.References.BID {
+		rs = append(rs, reference.Reference{
+			Source: "vuxml.freebsd.org",
+			// The URL i.e. http://www.securityfocus.com/bid/12615 is 503 at 2024-04-21,
+			// we should use, for example, WebArchive.org waybackmachine.
+			URL: fmt.Sprintf("http://www.securityfocus.com/bid/%s", b),
+		})
+	}
+	for _, c := range fetched.References.CertSA {
+		rs = append(rs, reference.Reference{
+			Source: "vuxml.freebsd.org",
+			// The URL http://www.cert.org/advisories/CA-2004-01.html is redirected to not very detailed page,
+			// Because there is only one certsa tag at 2004, leave it as it is.
+			URL: fmt.Sprintf("http://www.cert.org/advisories/%s.html", c),
+		})
+	}
+	for _, c := range fetched.References.CertVU {
+		rs = append(rs, reference.Reference{
+			Source: "vuxml.freebsd.org",
+			URL:    fmt.Sprintf("https://www.kb.cert.org/vuls/id/%s", c),
+		})
+	}
+	for _, u := range fetched.References.USCertTA {
+		rs = append(rs, reference.Reference{
+			Source: "vuxml.freebsd.org",
+			// The URL i.e. http://www.uscert.gov/cas/techalerts/TA07-199A.html is 503 at 2024-04-21,
+			// we should use, for example, WebArchive.org waybackmachine.
+			URL: fmt.Sprintf("http://www.uscert.gov/cas/techalerts/%s.html", u),
+		})
+	}
+
+	vs := make([]types.Vulnerability, 0, len(fetched.References.Cvename))
+	for _, c := range fetched.References.Cvename {
+		vs = append(vs, types.Vulnerability{
+			ID: c,
+			References: []reference.Reference{{
+				Source: "vuxml.freebsd.org",
+				URL:    fmt.Sprintf("https://nvd.nist.gov/vuln/detail/%s", c),
+			}}})
+	}
+
+	ds := make([]detection.Detection, 0, len(fetched.Affects))
+	for _, a := range fetched.Affects {
+		for _, n := range a.Name {
+			rs := make([]detection.Range, 0, len(a.Range))
+			for _, r := range a.Range {
+				rs = append(rs, detection.Range{Equal: r.Eq, LessThan: r.Lt, LessEqual: r.Le, GreaterThan: r.Gt, GreaterEqual: r.Ge})
+			}
+			ds = append(ds, detection.Detection{
+				Ecosystem:  detection.EcosystemTypeFreeBSD,
+				Vulnerable: true,
+				Package:    detection.Package{Name: n},
+				Affected: &detection.Affected{
+					Type:  detection.RangeTypeVersion,
```

Comment:
Please create a new RangeType for FreeBSD pkg.
```suggestion
					Type:  detection.RangeTypeFreeBSDPkg,
```

Reply:
Added!